### PR TITLE
AUTH-5315 adds new conditional_access_enabled attr to access IDP

### PR DESCRIPTION
**Pull request by ghost**
## Description

* adds new conditional_access_enabled attr to access IDP
* list/put IDP auth context endpoints
* auth context rule type

## Has your change been tested?

* Added unit tests for policy creation w/new rule and auth context endpoints

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This change is using publicly documented in [cloudflare/api-schemas](https://github.com/cloudflare/api-schemas) 
      and relies on stable APIs.

### CODE REVIEW

File: access_group.go [hunk 1]
```diff
@@ -153,6 +153,15 @@ type AccessGroupSAML struct {
 	} `json:"saml"`
 }
 
+// AccessGroupAzureAuthContext is used to configure access based on Azure auth contexts.
+type AccessGroupAzureAuthContext struct {
+	AuthContext struct {
```
Comment (jacobbednarz) [inline access_group.go:158]: can we use external structs here? inline structs are hard to use inline.

File: access_identity_provider.go [hunk 2]
```diff
@@ -235,3 +252,51 @@ func (api *API) DeleteAccessIdentityProvider(ctx context.Context, rc *ResourceCo
...
+// Zone API Reference: https://developers.cloudflare.com/api/operations/zone-level-access-identity-providers-update-an-access-identity-provider
+func (api *API) UpdateAccessIdentityProviderAuthContexts(ctx context.Context, rc *ResourceContainer, identityProviderID string) (AccessIdentityProvider, error) {
+	uri := fmt.Sprintf(
+		"/%s/%s/access/identity_providers/%s/auth_context",
+		rc.Level,
+		rc.Identifier,
+		identityProviderID,
+	)
+
+	res, err := api.makeRequestContext(ctx, http.MethodPut, uri, nil)
```
Comment (jacobbednarz) [inline access_identity_provider.go:290]: no payload?

File: access_group.go (hunk 1 above)
Comment (ghost) [reply access_group.go:158]: For code consistency I'd say not in this PR if possible. We currently define all our groups this way

File: access_identity_provider.go (hunk 2 above)
Comment (ghost) [reply access_identity_provider.go:290]: Yes this is similar to our refreshSCIM endpoint (although now that I'm looking maybe it's not in this repo yet). Essentially a PUT to /identity_providers/<ID>/auth_contexts will pull the updates from their IDP, they don't need to be provided in the request.

File: access_group.go (hunk 1 above)
Comment (jacobbednarz) [reply access_group.go:158]: ah you're right. we can look at that another time.

File: access_identity_provider.go [hunk 3]
```diff
@@ -89,6 +90,22 @@ type UpdateAccessIdentityProviderParams struct {
 	ScimConfig AccessIdentityProviderScimConfiguration `json:"scim_config"`
 }
 
+// AccessAuthContext represents an Access Azure Identity Provider Auth Context
```
Comment (jacobbednarz) [inline access_identity_provider.go]:
```suggestion
// AccessAuthContext represents an Access Azure Identity Provider Auth Context.
```

**Review by jacobbednarz — COMMENTED** [review]
lgtm once we have public docs 🚀

**Pull request merged**